skimage/metrics/simple_metrics.py

```diff
@@ -257,11 +261,18 @@ def normalized_mutual_information(image0, image1, *, bins=100):
     else:
         padded0, padded1 = image0, image1
 
-    hist, bin_edges = np.histogramdd(
+    if weights is not None and weights.shape != padded0.shape:
+        max_shape = np.maximum(image0.shape, image1.shape)
+        padded_weights = _pad_to(weights, max_shape)
+    else:
+        padded_weights = weights
+
+    hist = np.histogramdd(
         [np.reshape(padded0, -1), np.reshape(padded1, -1)],
         bins=bins,
         density=True,
-    )
+        weights=None if weights is None else np.reshape(padded_weights, -1),
+    )[0]
```

Comment:
Not sure, but isn't the following more natural to follow? 
```py
    if weights is not None:
        if weights.shape != padded0.shape:
            max_shape = np.maximum(image0.shape, image1.shape)
            padded_weights = _pad_to(weights, max_shape)
        else:
            padded_weights = weights
        weights = np.reshape(padded_weights, -1)

    hist = np.histogramdd(
        [np.reshape(padded0, -1), np.reshape(padded1, -1)],
        bins=bins,
        density=True,
        weights=weights,
    )[0]
```

Reply:
Yes, it is nicer.